src/python/pants/option/parser.py

```diff
@@ -611,23 +623,11 @@ def expand(val_or_str):
             flag_vals.append(
                 ListValueComponent(ListValueComponent.MODIFY, [*passthru_arg_strs], [])
             )
-
-        if is_list_option(kwargs):
```

Comment:
This now happens in `merge_in_rank` which is used by all vals, not just flag vals.